clients/src/NetworkConfig.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * Config representing the deployed state of bls-wallet contracts
+ */
+export type NetworkConfig = Readonly<{
+    /**
+     * Parameters used in contract deployment. Currently unused.
+     */
+    parameters: {},
+    /**
+     * Contract addresses
+     */
+    addresses: {
+        precompileCostEstimator: string,
+        verificationGateway: string,
+        blsLibrary: string,
+        blsExpander: string,
+        testToken: string,
+        rewardToken: string
+    },
+    /**
+     * Additional information about deployment/deployed state
+     */
+    auxiliary: {
+        chainid: number,
+        /**
+         * Domain used for BLS signing
+         */
+        domain: string,
+        /**
+         * Starting block contracts began dpeloyment at 
+         */
+        genesisBlock: number,
+        /**
+         * Address of the EOA which deployed the contracts
+         */
+        deployedBy: string,
+        /**
+         * Git commit SHA of the contracts
+         */
+        version: string
+    }
+}>;
+
+type ReadFileFunc = (filePath: string) => Promise<string>;
+
+/**
+ * Rerieves Network config
+ * 
+ * @param networkConfigPath Path to config
+ * @param readFileFunc Callback to retrieve the config. This could be via fetch, fs.readFile, etc.
+ */
+export async function getConfig(networkConfigPath: string, readFileFunc: ReadFileFunc): Promise<NetworkConfig> {
+    return JSON.parse(await readFileFunc(networkConfigPath));
```

Comment:
I'm against assigning types optimistically to functions that return `any`. If the json file doesn't match the type, it can break in confusing ways downstream, especially if there's type logic going on.

Have you ever used `io-ts`? That would be my go-to solution here.